src/actions/result.ts

```diff
@@ -18,22 +20,26 @@ export type ResultRequest = ReduxAction<typeof RESULT_REQUEST, {}>;
 
 export const RESULT_RECEIVE = 'RESULT_RECEIVE';
 export type ResultReceive = ReduxAction<typeof RESULT_RECEIVE, {
-  modelGroup: SpecQueryModelGroup
+  modelGroup: SpecQueryGroup<PlotObject>
 }>;
 
 export type AsyncResultRequest = ThunkAction<void , State, undefined>;
-export function resultRequest(query?: Query, schema?: Schema): AsyncResultRequest {
+export function resultRequest(query?: Query, schema?: Schema, data?: Data): AsyncResultRequest {
   return (dispatch: Dispatch<Action>, getState) => {
     if (!query) {
       query = getQuery(getState());
     }
     if (!schema) {
       schema = getSchema(getState());
     }
+    if (!data) {
```

Comment:
good point. I updated and removed them from the function signature. originally, the idea was to use this in multiple places - but based on how we are calling it now, it is not necessary. thanks!